packages/intent-filler/src/core/filler.ts

```diff
@@ -0,0 +1,76 @@
+import { EventMonitor } from "./event-monitor"
+import { FillerStrategy } from "@/strategies/base"
+import { Order, FillerConfig, ChainConfig } from "@/types"
+import pQueue from "p-queue"
+
+export class IntentFiller {
+	private monitor: EventMonitor
+	private strategies: FillerStrategy[]
+	private config: FillerConfig
+	private orderQueue: pQueue
+
+	constructor(chainConfigs: ChainConfig[], strategies: FillerStrategy[], config: FillerConfig) {
+		this.monitor = new EventMonitor(chainConfigs)
+		this.strategies = strategies
+		this.config = config
+
+		// Create a concurrent queue for processing orders
+		this.orderQueue = new pQueue({
```

Comment:
I think we should have a queue for each chain, transactions to the same chain have to be executed sequentially since EVM does not allow concurrent transaction submission for an account.

Reply:
Missed it. Thanks for catching.